sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -70,7 +71,7 @@ private[sql] case class InsertIntoHadoopFsRelation(
       relation.paths.length == 1,
       s"Cannot write to multiple destinations: ${relation.paths.mkString(",")}")
 
-    val hadoopConf = sqlContext.sparkContext.hadoopConfiguration
+    val hadoopConf = new Configuration(sqlContext.sparkContext.hadoopConfiguration)
```

Comment:
Do we need this? We already do `val job = new Job(hadoopConf)` below. BTW, we need to add comment to explain `new Job` will clone the conf.